### PR TITLE
changefeedccl: Issue high priority reads when polling descriptors.

### PR DESCRIPTION
**Pull request by miretskiy**
Changefeeds poll system descriptors periodically, once every second by default.
However, if there is an intent on the descriptors table that is not getting cleared
out, then changefeeds would appear to be stuck.

Add a mechanism to schema feed whereby it will issue a high priority read
if the regular priority read does not succeed within threshold specified
via `changefeed.schema_feed.read_with_priority_after` cluster setting.

Release Notes: None
Release Justification: Stability improvement.

### CODE REVIEW

File: pkg/ccl/changefeedccl/schemafeed/schema_feed.go [hunk 1]
```diff
@@ -609,13 +618,56 @@ func (tf *schemaFeed) fetchDescriptorVersions(
...
+		header.UserPriority = roachpb.MaxUserPriority
+		return fetchDescriptors(ctx)
+	}
+	return nil, err
+}
+
+func (tf *schemaFeed) fetchDescriptorVersions(
+	ctx context.Context, startTS, endTS hlc.Timestamp,
+) ([]catalog.Descriptor, error) {
 	if log.V(2) {
```
Comment (ajwerner) [inline pkg/ccl/changefeedccl/schemafeed/schema_feed.go:658]: can you change this to `log.ExpensiveLogEnabled(ctx, 2)` and below

**Review by ajwerner — APPROVED** [review]
As we discussed, this isn't needed, but it seems fine. It's good to have a test of the effect, but it's a little weird to not test the actual logic. Maybe one think you could do is plumb a fake codec through and use the kv DB to write an intent to that descriptor table.

**Review by ajwerner — APPROVED** [review]
Missed the override's impact. Lgtm

**Review by HonoreDB — COMMENTED** [review]
> Reviewed 1 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/79575)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @HonoreDB and @miretskiy)

___
*[pkg/ccl/changefeedccl/schemafeed/schema_feed.go, line 648 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/79575#-N-3Js1o0-J4RXJsJkB8:-N-3Js1o0-J4RXJsJkB9:b-g7zq9m) ([raw file](https://github.com/cockroachdb/cockroach/blob/b58f2b1c9423fdc029638837e7c7e4eb5e5baa7c/pkg/ccl/changefeedccl/schemafeed/schema_feed.go#L648)):*
> ```Go
> 		return resp, nil
> 	}
> 	if errors.HasType(err, (*contextutil.TimeoutError)(nil)) {
> ```

` && header.UserPriority != roachpb.MaxUserPriority`?

**Comment by miretskiy** [comment]
> && header.UserPriority != roachpb.MaxUserPriority?
Why? We only retry once with max prio...

**Review by HonoreDB — APPROVED** [review]

**Pull request merged**